test/micro-conf-validation.test.ts

```diff
@@ -0,0 +1,283 @@
+	process.env.NODE_ENV = 'development';
+	const file = await tmp.file();
+	const microPath = `${microConfValidation}/configuration-valid-with-additional-attributes`;
+
+	const { server } = await N9NodeRouting({
+		path: microPath,
+		logOptions: { developmentOutputFilePath: file.path },
+		conf: {
+			n9NodeConf: {
+				path: `${microPath}/conf`,
+			},
+			validation: {
+				isEnabled: true,
+				classType: ValidConfWithWhitelistErrors,
+				formatWhitelistErrors: true,
+			},
+		},
+	});
+
+	const output = await getLogsFromFile(file.path);
+
+	t.true(output[1].includes('Checking configuration'));
+	t.true(output[2].includes('Configuration contains unexpected attributes'));
+	t.true(output[3].includes("Please remove attribute 'whitelist1'"));
+	t.true(output[4].includes("Please remove attribute 'whitelist2'"));
+	t.true(output[5].includes("Please remove attribute 'baz.qux'"));
+	t.true(output[6].includes('Configuration is valid'));
+	t.true(output[7].includes('Listening on port'));
+
+	await closeServer(server);
+});
+
+ava(
+	'Should be a valid configuration with whitelist errors (not formatted)',
+	async (t: Assertions) => {
+		process.env.NODE_ENV = 'development';
+		const file = await tmp.file();
+		const microPath = `${microConfValidation}/configuration-valid-with-additional-attributes`;
+
+		const { server } = await N9NodeRouting({
+			path: microPath,
+			logOptions: { developmentOutputFilePath: file.path },
+			conf: {
+				n9NodeConf: {
+					path: `${microPath}/conf`,
+				},
+				validation: {
+					isEnabled: true,
+					classType: ValidConfWithWhitelistErrors,
+					formatWhitelistErrors: false,
+				},
+			},
+		});
+
+		const output = await getLogsFromFile(file.path);
+
+		t.true(output[1].includes('Checking configuration'));
+
+		t.true(output[2].includes('Configuration contains unexpected attributes'));
+		t.true(output[2].includes('property qux should not exist'));
+		t.true(output[2].includes('property whitelist1 should not exist'));
+		t.true(output[2].includes('property whitelist2 should not exist'));
+
+		t.true(output[3].includes('Configuration is valid'));
+		t.true(output[4].includes('Listening on port'));
+
+		await closeServer(server);
+	},
+);
+
+ava('Should not proceed any validation - no validation options given', async (t: Assertions) => {
+	process.env.NODE_ENV = 'development';
+	const file = await tmp.file();
+	const microPath = `${microConfValidation}/configuration-invalid`;
+
+	const { server } = await N9NodeRouting({
+		path: microPath,
+		logOptions: { developmentOutputFilePath: file.path },
+		conf: {
+			n9NodeConf: {
+				path: `${microPath}/conf`,
+			},
+		},
+	});
+
+	const output = await getLogsFromFile(file.path);
+
+	t.true(output[1].includes('Configuration validation is disabled'));
+	t.true(output[2].includes('Listening on port'));
+
+	await closeServer(server);
+});
+
+ava('Should not proceed any validation - validation is disabled', async (t: Assertions) => {
```

Comment:
Could you add the same test but with deactivation of the validation from the configuration itself, in configuration extension for example ?